compiler/rustc_codegen_llvm/src/asm.rs

```diff
@@ -874,3 +913,31 @@ fn llvm_fixup_output_type(
         _ => layout.llvm_type(cx),
     }
 }
+
+/// Gets a LLVM label from a &str if it exists
+/// Uses this as reference: https://llvm.org/docs/AMDGPUOperandSyntax.html#symbols
+fn get_llvm_label_from_str(s: &str) -> Option<&str> {
+    if let Some(colon_idx) = s.find(':') {
+        let substr = &s[..colon_idx];
+        let mut chars = substr.chars();
+        if let Some(c) = chars.next() {
+            // First char is [a-zA-Z_.]
+            if ('a'..='z').contains(&c) || ('A'..='Z').contains(&c) || '_' == c || '.' == c {
+                // All subsequent chars are [a-zA-Z0-9_$.@]
+                if chars.all(|c| {
+                    ('a'..='z').contains(&c)
+                        || ('A'..='Z').contains(&c)
+                        || ('0'..='9').contains(&c)
+                        || '_' == c
+                        || '$' == c
+                        || '.' == c
+                        || '@' == c
+                }) {
```

Comment:
```suggestion
                if chars.all(valid_ident_continuation) {
```
This is the same as the new function defined above (if I'm not misled).

Reply:
Yes this is meant to be the same.  The function above is local though and was added later.  I think I'll split out the llvm identifier start and continuation functions and use them.